### ISSUE TITLE
attempt to call field 'isarray' (a nil value) - some kind of configuration problem 

### ISSUE THREAD

**Issue by goolord** [Shatur/neovim-session-manager#144]
<img width="862" alt="image" src="https://github.com/user-attachments/assets/75487785-9e9d-4875-8180-8d9812894157">
when using the lazy package manager, I get this error when `lazy = true` isn't specified.

I guess it's some problem with when the `SessionLoad*` event is field and when the code that defines `vim.isarray` is evaluated, but I'm unsure. I would maybe just consider not using `vim.isarray`, but feel free to just close this if you don't want to support the insane number of permutations neovim configurations come in. 

thought this might be helpful to other end users anyways

**Comment by Shatur** [comment]
Duplicate of #127.

**Comment by goolord** [comment]
not a duplicate - this is not caused by erroneous configuration or an outdated version of neovim. it's a similar error caused by a separate problem

**Comment by Shatur** [comment]
The issue in this line:
https://github.com/Shatur/neovim-session-manager/blob/ce43f2eb2a52492157d7742e5f684b9a42bb3e5c/lua/session_manager/init.lua#L122
Something is wrong with either your version of Neovim or `config.autoload_mode`.

**Comment by goolord** [comment]
```lua
    {
        'Shatur/neovim-session-manager',
        config = function()
            -- vim.isarray = function () return false end
            require('session_manager').setup {
                autoload_mode = require('session_manager.config').AutoloadMode.CurrentDir,
                autosave_last_session = true,
            }
        end,
        dependencies = {
            'nvim-lua/plenary.nvim',
        }
    };
```
looks completely fine to me, does not error if I define vim.isarray

**Comment by Shatur** [comment]
Weird, not sure why it happens...

**Comment by Njima1572** [comment]
I encountered the same problem, and it could be because of using nvim v10.0-dev
This version of neovim doesn't seem to have `vm.isarray`, but instead has `vim.tbl_isarray`

https://github.com/neovim/neovim/commit/7caf0eafd83b5a92f2ff219b3a64ffae4174b9af

**Comment by goolord** [comment]
yes it was a problem with my neovim version. in my case it was opaque to me because somehow the command `nvim --version` and the command `nvim` were resolving to two different neovim binaries
